src/generate-query.ts

```diff
@@ -610,6 +669,145 @@ function getArgsAndVars(
   }
 }
 
+function getListSizeDirective(field: FieldDefinitionNode): DirectiveNode {
+  return field?.directives.find((dir) => dir.name.value === 'listSize')
+}
+
+function getSlicingArguments(
```

Comment:
I think this function is a bit strange. It is doing two things, 1) getting the list size directive (if it isn't provided and the `field` is) and 2) extracting the slicing arguments. I think the two functions should be split apart. We should have a `getListSizeDirective()` function (which we already have) and a `getSlicingArgumentsFromListSIzeDirective()` function that just takes a list size directive. Then in the code, if we already have the list size directive, we just call the latter, and if we only have a field, then we call both.

Reply:
Yes, that makes sense. I will change the code as you mentioned.